core/src/main/java/io/grpc/ProxyParameters.java

```diff
@@ -58,4 +81,43 @@ public boolean equals(Object o) {
   public int hashCode() {
     return Objects.hashCode(proxyAddress, username, password);
   }
+
+  /**
+   * Create a new builder.
+   */
+  public static Builder forAddress(SocketAddress proxyAddress) {
+    return new Builder(proxyAddress);
+  }
+
+  /**
+   * The helper class to build an Attributes instance.
+   */
+  public static final class Builder {
+
+    private final SocketAddress proxyAddress;
+    @Nullable
+    private String username;
+    @Nullable
+    private String password;
+
+    private Builder(SocketAddress proxyAddress) {
+      this.proxyAddress = Preconditions.checkNotNull(proxyAddress, "proxyAddress");
+    }
+
+    @Nullable
+    public Builder username(String username) {
```

Comment:
Arg my other comment is showing the wrong code. Re-creating to show `@Nullable` is in wrong spot. Ditto below.

Reply:
Oh, yes, I missed that. Fixed